sim-administration/es2plus4dropwizard/src/main/kotlin/org/ostelco/sim/es2plus/Es2PlusClient.kt

```diff
@@ -30,90 +30,116 @@ class ES2PlusClient(
         private val useHttps: Boolean = true,
         private val jerseyClient: Client? = null) {
 
+    // TODO: Make a new interface to represent the client we are using, then set a non-nullable field
+    //       (private) to contain one instande that represents either a HttpClient or a jerseyClient
+    //       wrapped in the appropriate protocol logic.   This will get rid of the silly ==null tests
+    //       that riddle this class.
+
+
```

Comment:
Line 29 (outside of changeset, but...). Would it not be better to not allow `httpClient` to be `null`?

Reply:
OK. You can use either a Jersey or a http client (line 29 and 31). Maybe just settle for one?

Reply:
Agree with the TODO!

Reply:
The reason I have both is that the jersey client does not work well with SSL, and the other one does (after a fashion, at least it works). However, the jersey client has excellent support  for unit testing, and most of the original code was written using jersey client,, and it was very productive to do so.   I will look into fixing this one way or another, since having two clients present in this somewhat messy way is not a good practice,  but I will not do that in this PR.